apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.cs

```diff
@@ -333,48 +334,69 @@ public override Task<int> ExecuteNonQueryAsync(CancellationToken cancellationTok
             CancellationToken cancellationToken)
         {
             var databaseAdminClient = await DatabaseAdminClient.CreateAsync().ConfigureAwait(false);
-
-            if (spannerCommandTextBuilder.IsCreateDatabaseCommand)
-            {
-                var parent = new InstanceName(spannerConnection.Project, spannerConnection.SpannerInstance);
-                var response = await databaseAdminClient.CreateDatabaseAsync(
-                    new CreateDatabaseRequest
-                    {
-                        ParentAsInstanceName = parent,
-                        CreateStatement = commandText,
-                        ExtraStatements = { spannerCommandTextBuilder.ExtraStatements }
-                    }).ConfigureAwait(false);
-                await response.PollUntilCompletedAsync().ConfigureAwait(false);
-            }
-            else if (spannerCommandTextBuilder.IsDropDatabaseCommand)
+            try
             {
-                if (spannerCommandTextBuilder.ExtraStatements?.Length > 0)
+                if (spannerCommandTextBuilder.IsCreateDatabaseCommand)
                 {
-                    throw new InvalidOperationException("Drop database commands do not support additional ddl statements");
+                    var parent = new InstanceName(spannerConnection.Project, spannerConnection.SpannerInstance);
+                    var response = await databaseAdminClient.CreateDatabaseAsync(
+                        new CreateDatabaseRequest
+                        {
+                            ParentAsInstanceName = parent,
+                            CreateStatement = commandText,
+                            ExtraStatements = {spannerCommandTextBuilder.ExtraStatements}
+                        }).ConfigureAwait(false);
+                    response = await response.PollUntilCompletedAsync().ConfigureAwait(false);
+                    if (response.IsFaulted)
+                    {
+                        throw new SpannerException(
+                            ErrorCode.Unknown, response.Exception?.Message ?? "An unknown Rpc error occurred.");
```

Comment:
I'm not familiar with most of this code but can you pull a response code from response or response.Exception if it may have more info than "Unknown"?

Reply:
Hmm many of them do map to grpc codes.  It's probably worth doing.